src/plugins/intel_gpu/include/intel_gpu/runtime/utils.hpp

```diff
@@ -255,20 +256,27 @@ class LRUCache {
     }
 
 private:
-    void add_new_entry(TypeK key, LRUCache::CacheEntry entry) {
-        if (cache_entry_map.find(key) == cache_entry_map.end()) {
-            if (curr_data_size > 0 && capacity < (curr_data_size + entry.size)) {
-                //  Remove cache at the end of order
-                curr_data_size -= cache_entry_map[order.back()].second.size;
-                cache_entry_map.erase(order.back());
-                order.pop_back();
-            }
-        } else {
+    bool add_new_entry(TypeK key, LRUCache::CacheEntry entry) {
+        if (entry.size > capacity)
```

Comment:
How about showing a warning message saying the entry could not be added to the cache for the size issue using GPU_DEBUG ?

` GPU_DEBUG_GET_INSTANCE(debug_config);
GPU_DEBUG_IF(debug_config->verbose >= 3) {}`